packages/astro/src/assets/utils/node/emitAsset.ts

```diff
@@ -9,6 +10,17 @@ import { imageMetadata } from '../metadata.js';
 type FileEmitter = vite.Rollup.EmitFile;
 type ImageMetadataWithContents = ImageMetadata & { contents?: Buffer };
 
+// Global cache for SVG content deduplication
+// Maps content hash -> { handle: string, filename: string }
+const svgContentCache = new Map<string, { handle: string; filename: string }>();
```

Comment:
Since this is a cache and we don't do any iteration, I think we can use a [`WeakMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) instead. However, a `WeakMap` needs an object as a key. So maybe we can create a new object e.g. `SvgCacheKey = { hash: string, fileData: Buffer }`. Not sure if we need `fileData`, so try to experiment